insights/parsers/parted.py

```diff
@@ -231,3 +168,189 @@ def parse_content(self, content):
+        >>> sorted(parted_info.data.items())
+        [('disk', '/dev/sda'), ('disk_flags', 'pmbr_boot'), ('model', 'ATA TOSHIBA MG04ACA4 (scsi)'), ('partition_table', 'gpt'), ('sector_size', '512B/512B'), ('size', '4001GB')]
+        >>> parted_info.data['model']
+        'ATA TOSHIBA MG04ACA4 (scsi)'
+        >>> parted_info.disk
+        '/dev/sda'
+        >>> parted_info.logical_sector_size
+        '512B'
+        >>> parted_info.physical_sector_size
+        '512B'
+        >>> parted_info.boot_partition
+        >>> parted_info.data['disk_flags']
+        'pmbr_boot'
+        >>> len(parted_info.partitions)
+        3
+        >>> sorted(parted_info.partitions[0].data.items())
+        [('end', '2097kB'), ('file_system', ''), ('flags', 'bios_grub'), ('name', ''), ('number', '1'), ('size', '1049kB'), ('start', '1049kB')]
+        >>> parted_info.partitions[0].number
+        '1'
+        >>> parted_info.partitions[0].start
+        '1049kB'
+        >>> parted_info.partitions[0].end
+        '2097kB'
+        >>> parted_info.partitions[0].size
+        '1049kB'
+        >>> parted_info.partitions[0].file_system
+        ''
+        >>> parted_info.partitions[0].type
+        >>> parted_info.partitions[0].flags
+        'bios_grub'
+
+    """
+
+    def get(self, disk_name):
+        """Returns a value for the specified ``item`` key."""
+        for dev in self.devices_info:
+            if dev.disk == disk_name:
+                return dev
+
+    def __iter__(self):
+        for dev in self.devices_info:
+            yield dev
+
+    def parse_content(self, content):
+        """Divide "content" into each device section by two empty lines."""
+
+        device_tables = []
+
+        # Divide "content" into each Device Section by two empty lines.
+        idx, this_start_idx = 0, 0
+        is_last_line_empty = False
+        for idx, raw_line in enumerate(content):
+            if len(raw_line.strip()) == 0:
+                if is_last_line_empty:
+                    # Fount this line as the second blank line
+                    device_tables.append(content[this_start_idx:idx - 1])
+                    this_start_idx = idx + 1
+                    is_last_line_empty = False
+                else:
+                    is_last_line_empty = True
+            else:
+                is_last_line_empty = False
+
+        # Take left "content" as the last Device Section
+        device_tables.append(content[this_start_idx:])
+
+        devices_info = []
+        for dev_table in device_tables:
+            try:
+                this_parsed_dev_table = PartedDevice(dev_table)
+                devices_info.append(this_parsed_dev_table)
+            except ParseException:
+                # Mute any exception from PartedDevice parsing.
+                # Keep only fully parsed Device into the final devices_info.
+                pass
+
+        self.devices_info = devices_info
+
+
+@parser(Specs.parted_s_sos)
+class PartedSos(CommandParser):
```

Comment:
I didn't find there are any GSS rules that used the original "Parted" parser , so it's not so necessary to add this one for SOS here on the core side. I mean, we can remove this one, and later if there are such requests from GSS side, we can also add it to the GSS models.

Reply:
Removed the `PartedSos`.